src/main/pages/che-7/administration-guide/proc_configuring-prod-short-to-use-external-keycloak.adoc

```diff
@@ -0,0 +1,77 @@
++
+image::keycloak/external_keycloak_realm.png[link="{imagesdir}/keycloak/external_keycloak_realm.png]
+
+
+* In this `realm`, define an link:https://www.keycloak.org/docs/latest/server_admin/#oidc-clients[OIDC client] that {prod-short} will use to authenticate the users. The following steps use the example `public-client` name. Here is an example of such a client with the correct settings:
++
+image::keycloak/external_keycloak_public_client.png[link="{imagesdir}/keycloak/external_keycloak_public_client.png]
+
+
+[CAUTION]
+====
+* {prod-short} only supports `public` OIDC clients. Therefore, selecting the `openid-connect` `Client Protocol` option and the `public` Access Type option is highly recommended.
+
+* The list of `Valid Redirect URIs` must contain at least 2 URIs related to the {prod-short} server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, followed by `/*` wildcards.
+
+* The list of `Web Origins` must contain at least 2 URIs related to the Che server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, without any path after the host.
++
+The number of URIs depends on the number of installed product tools.
+====
+
+* If {prod-short} is installed
+ifeval::["{project-context}" == "che"]
+on OpenShift
+endif::[]
+and uses the default OpenShift OAuth support, user authentication relies on the integration of {identity-provider} with OpenShift OAuth. This allows users to log in to {prod-short} with their OpenShift login and have their workspaces created under personal OpenShift namespaces. 
++
+This requires setting up an OpenShift identity provider inside {identity-provider}. When using an external {identity-provider}, set up the identity provider manually. For instructions, seethe appropriate {identity-provider} documentations for either link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/html/server_administration_guide/identity_broker#openshift_3[OpenShift 3] or link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/html/server_administration_guide/identity_broker#openshift_4[OpenShift 4].
+
+* The configured identity provider has `Store Tokens` and `Stored Tokens Readable` options enabled.
+
+
+.Procedure
+
+. Set the following properties in the `CheCluster` Custom Resource (CR):
++
+[source, yaml]
+----
+spec:
+  auth:
+    externalIdentityProvider: true
+    identityProviderURL: <Base URL of your external Keycloak server>
+    # CAUTION: realm-for-users should be replaced by the name of your realm
+    identityProviderRealm: realm-for-users
+    # CAUTION: public-client should be replaced by the name of your client
+    identityProviderClientId: public-client
```

Comment:
Our standard I've seen is to define a placeholder at the top, state what it is, and then use it throughout, e.g. `<auth-realm>` and `<client-id>`